### PR TITLE
AO3-6371 Add no resets role

### PR DESCRIPTION
**Pull request by sarken**
## Issue

https://otwarchive.atlassian.net/browse/AO3-6371

## Purpose

Maims the English language in service of creating a user role that _only_ prevents password resets, not other forms of harassment.

## Testing Instructions

Refer to Jira.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -416,6 +419,12 @@ def is_protected_user?
     has_role?(:protected_user)
   end
 
+  # Is this user assigned the no resets role? These users do no wish to receive
+  # password resets.
+  def is_no_resets?
```
[the remark this answers is not in the document]
Comment (sarken) [reply app/models/user.rb:424]: Grammatically, this is better. But it is not what we do for roles.

[the remark this answers is not in the document]
Comment (redsummernight) [reply app/models/user.rb:424]: Let's start using the better option now. You can ignore the existing names though.

File: features/users/authenticate_users.feature [hunk 2]
```diff
@@ -244,3 +244,22 @@ Feature: User Authentication
     Then I should be on the home page
       And I should see "Password resets are disabled for that user."
       And 0 emails should be delivered
+
+  Scenario: Passwords cannot be reset for users who have been given the no resets role.
```
Comment (redsummernight) [inline features/users/authenticate_users.feature:248]: You can combine tests for the two roles into a `Scenario Outline`?

File: app/models/user.rb [hunk 3]
```diff
@@ -416,6 +419,12 @@ def is_protected_user?
     has_role?(:protected_user)
   end
 
+  # Is this user assigned the no resets role? These users do no wish to receive
```
Comment (redsummernight) [inline app/models/user.rb:422]: Should be "do not wish".

**Review by redsummernight — APPROVED** [review]

**Pull request merged**